pkg/config/sync.go

```diff
@@ -71,14 +71,13 @@ func SetupControllerConfig(client crclient.Client) error {
 	} else {
 		syncConfigFrom(config)
 	}
+	defaultRoutingSuffix, err := discoverRouteSuffix(client)
+	if err != nil {
+		return err
+	}
+	DefaultConfig.Routing.ClusterHostSuffix = defaultRoutingSuffix
 	if internalConfig.Routing.ClusterHostSuffix == "" {
```

Comment:
I just wonder if there is a way to avoid evaluation ClusterHostSuffix on each reconcile loop?
It may be overhead but we may use configStatus for such stuff.